### PR TITLE
Avoid load stuff not required after reset

### PR DESCRIPTION
**Pull request by Christian-B**
There was a considerable amount of stuff that was done after reset even when the application graph had not changed.

Interestingly enough one thing not done was to change the report folder.
Which meant some reports went pop when trying to recreate the same sub directory.

### CODE REVIEW

**Comment by alan-stokes** [comment]
request to remove self._application_graph_changed from a class variable to just a local. the risks of keeping it updated are more dangerous as class variable

**Comment by alan-stokes** [comment]
1705 merge the ifs

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 1]
```diff
@@ -1677,20 +1677,34 @@ def _do_load(self):
...
         # add report for extracting routing table from machine report if needed
         # Add algorithm to clear routing tables and set up routing
-        if not self._use_virtual_board:
-            algorithms.append("RoutingSetup")
-            # Get the executable targets
-            algorithms.append("GraphBinaryGatherer")
+        if not self._use_virtual_board and application_graph_changed:
+                algorithms.append("RoutingSetup")
+                # Get the executable targets
+                algorithms.append("GraphBinaryGatherer")
```
Comment (dkfellows) [inline spinn_front_end_common/interface/abstract_spinnaker_base.py]: Bonus indentation? Will flake8 like it?

Comment (alan-stokes) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py]: anwser = no

Comment (alan-stokes) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py]: does flake8 like anything

Comment (dkfellows) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py]: Meh. It's allowed, but it still sucks slightly.

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 2]
```diff
@@ -1677,20 +1677,34 @@ def _do_load(self):
...
             algorithms.append("GraphBinaryGatherer")
 
-        if helpful_functions.read_config(
-                self._config, "Mapping", "loading_algorithms") is not None:
+        loading_algorithm = helpful_functions.read_config(
+            self._config, "Mapping", "loading_algorithms")
+        if loading_algorithm is not None and application_graph_changed:
             algorithms.extend(
                 self._config.get("Mapping", "loading_algorithms").split(","))
         algorithms.extend(self._extra_load_algorithms)
```
Comment (dkfellows) [inline spinn_front_end_common/interface/abstract_spinnaker_base.py]: Why are we not using the variable that we created a few lines above?

Comment (dkfellows) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py]: `loading_algorithm` that is…

Comment (dkfellows) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py]: We could just do `algorithms.extend(loading_algorithm.split(","))` and all would be good.

Comment (alan-stokes) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py]: good catch. i didnt notice when i did the refactor

**Pull request merged**